mitodl/open-discussions
# Added server-side 404 page for posts

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2785 

#### What's this PR do?
This adds a server-side 404 page for removed posts for non-moderator users, to ensure that for search engines and anyone else don't see this content perpetually. Also covered 400/403 errors just in case since they were a quick win.

#### How should this be manually tested?
- Remove a post in a public and/or private channel
- You should still be able to view the post as a moderator
- As a logged in user you should get a 404 page
- As an anonymous user you should get a 404 page for the public post

#### Any background context you want to provide?

Intentionally not handled here:

- 500 error pages: not in scope and a bit of effort
- Anonymous users trying to view a removed private post
